tests/infrastructure/golden_images/update_boot_source/test_boot_sources_vm.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+import pytest
+
+from utilities.infra import assert_os_version_mismatch_in_vm, validate_os_info_vmi_vs_linux_os
+from utilities.storage import data_volume_template_with_source_ref_dict
+from utilities.virt import VirtualMachineForTests, running_vm
+
+LOGGER = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def auto_update_boot_source_instance_type_vm(
+    unprivileged_client,
+    namespace,
+    modern_cpu_for_migration,
+    existing_data_source_volume,
+):
+    LOGGER.info(f"Create a VM using {existing_data_source_volume.name} dataSource")
+    with VirtualMachineForTests(
+        client=unprivileged_client,
+        name=f"{existing_data_source_volume.name}-data-source-vm",
+        namespace=namespace.name,
+        vm_instance_type_infer=True,
+        vm_preference_infer=True,
+        data_volume_template=data_volume_template_with_source_ref_dict(
+            data_source=existing_data_source_volume,
+        ),
+        cpu_model=modern_cpu_for_migration,
+    ) as vm:
+        running_vm(vm=vm)
+        yield vm
+
+
+@pytest.mark.polarion("CNV-7586")
+def test_instance_type_vm_from_auto_update_boot_source(
+    latest_fedora_release_version,
+    auto_update_boot_source_instance_type_vm,
+    boot_source_data_from_data_source_dict,
+):
+    LOGGER.info(f"Verify {auto_update_boot_source_instance_type_vm.name} OS version and virtctl info")
+    assert_os_version_mismatch_in_vm(
+        vm=auto_update_boot_source_instance_type_vm,
+        boot_source_data=boot_source_data_from_data_source_dict,
+        fedora_version=latest_fedora_release_version,
```

Comment:
Why would you pass fedora_version for non-fedora source?

Reply:
+1
i think you should do it here like in similar common template test